### PR TITLE
Add form type support in FormTrait for flexibility

### PR DESCRIPTION
**Pull request by lokhman**
Set parameter to the end for backwards compatibility, ideally make it first in the future release.

### CODE REVIEW

**Comment by fabpot** [comment]
That can only be merged in the master branch.

**Comment by lokhman** [comment]
Yes, that's my bad. Shall I close this request and open the new one for master commit then?

**Comment by fabpot** [comment]
@lokhman No, I can merge this PR directly into master.

**Comment by lokhman** [comment]
@fabpot Brilliant, thanks! Will it be possible to make `$type` argument coming first at some point in the future release? I personally find it rather important when extending `form.types` with own forms.

File: src/Silex/Application/FormTrait.php [hunk 1]
```diff
@@ -25,11 +25,12 @@
      *
      * @param mixed $data    The initial data for the form
      * @param array $options Options for the form
+     * @param mixed $type    Type of the form
```
Comment (stof) [inline src/Silex/Application/FormTrait.php]: why ``mixed`` ? The only supported types are ``FormTypeInterface`` and ``string``

Comment (lokhman) [inline src/Silex/Application/FormTrait.php]: True. Fixed it now to more correct.

**Comment by fabpot** [comment]
merged into master, thanks.

**Pull request closed without merging**